### ISSUE TITLE
Error in handleMapBrowserEvent for keydown event with map in shadowroot

### ISSUE THREAD

**Issue by MoonE** [openlayers/openlayers#12029]
**Describe the bug**
An error is thrown If the map is inside a shadow root element and a `keydown` event occurs inside with the map having focus.

The `clientX` and `clientY` properties are not available for key events.
https://github.com/openlayers/openlayers/blob/27b45393607bb25e03a82e4e35a2a065b450c3b0/src/ol/PluggableMap.js#L990-L1005
```
Uncaught TypeError: ShadowRoot.elementFromPoint: Argument 1 is not a finite floating-point value.
    handleMapBrowserEvent PluggableMap.js:1001
    handleBrowserEvent PluggableMap.js:976
```

**Comment by ahocevar** [comment]
Should we just take the event's target when the event is a keydown event?
```js
'host' in rootNode && eventType !== EventType.KEYDOWN
```
I don't think the target is critical for keydown events in OpenLayers.

**Comment by ahocevar** [comment]
I created a fix in #12083, using a completely different approach.
